runners/apex/src/test/java/org/apache/beam/runners/apex/examples/WordCountTest.java

```diff
@@ -123,11 +123,15 @@ public void testWordCountExample() throws Exception {
     options.setInputFile(new File(inputFile).getAbsolutePath());
     String outputFilePrefix = "target/wordcountresult.txt";
     options.setOutput(outputFilePrefix);
-    WordCountTest.main(TestPipeline.convertToArgs(options));
 
     File outFile1 = new File(outputFilePrefix + "-00000-of-00002");
     File outFile2 = new File(outputFilePrefix + "-00001-of-00002");
-    Assert.assertTrue(outFile1.exists() && outFile2.exists());
+    Assert.assertTrue(!outFile1.exists() || outFile1.delete());
```

Comment:
This catches the issue, but how about using `TemporaryFolder`?

Example: https://github.com/apache/beam/blob/084a5e8aedfdd53f3d76ad48b55a66a282881646/examples/java/src/test/java/org/apache/beam/examples/DebuggingWordCountTest.java#L36